### PR TITLE
add npe2pm TestPluginManager fixture

### PR DESCRIPTION
**Pull request by tlambert03**
This makes the pattern that napari is using for testing available to anyone as a pytest fixture named `npe2pm`.

~(need to add tests)~

### CODE REVIEW

File: npe2/_pytest_plugin.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        package : str
+            Name of an installed plugin/package.
+        name : str
+            If neither `manifest` or `package` is provided, a new DynamicPlugin is
+            created with this name, by default "tmp_plugin"
+
+        Returns
+        -------
+        DynamicPlugin
+            be sure to enter the DynamicPlugin context to register the plugin.
```
Comment (DragaDoncila) [inline npe2/_pytest_plugin.py:48]: maybe an example here?

**Review by DragaDoncila — CHANGES_REQUESTED** [review]
Lookin great! Just two small nit picks and a question for you

File: tests/test_pytest_plugin.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+def test_make_viewer(npe2pm):
+    with pytest.raises(FileNotFoundError):
+        npe2pm.tmp_plugin(manifest='some_path.yaml')
+"""
+
+
+@pytest.mark.parametrize("case", [CASE1, CASE2, CASE3, CASE4, CASE5, CASE6])
+def test_npe2pm_fixture(pytester: pytest.Pytester, case):
+    """Make sure that our make_napari_viewer plugin works."""
```
[the remark this answers is not in the document]
Comment (DragaDoncila) [reply tests/test_pytest_plugin.py:61]: Phew... I was starting to question reality for a second there...

File: npe2/_pytest_plugin.py (hunk 1 above)
Comment (tlambert03) [reply npe2/_pytest_plugin.py:48]: added

**Pull request merged**